### ISSUE TITLE
Create new v1.1 doc set in kubernetes.io

### ISSUE THREAD

**Issue by RichieEscarez** [kubernetes/kubernetes#15893]
- [x] 1. Create hidden v1.1 content:
     - [x] Create new .../v1.1/... directory on gh-pages #15987
         - [x] Create new .../v1.1/index.html (v1.1 doc banner) #15987
     - [x] Create new _includes/nav_v1.1.html file and add all topics into menu tree (navigation menu) #15903 
     - [x] Add drop-down version selector #16392
     - [x] Create and include a v1.0 banner: "This is an archived version of the documentation, the latest version is available here." #16163 
     - [x] Add "Disallow: /v1.0/" to existing [robot.txt](https://github.com/kubernetes/kubernetes/blob/gh-pages/robots.txt) file (to hide the v1.0 content from web crawlers and focus SEO on new v1.1) #16388
 - [x] 2. Run after all docs for v1.1 have been uploaded to gh-pages:
     - [x] Update existing [sitemap.xml](https://github.com/kubernetes/kubernetes/blob/gh-pages/sitemap.xml) file (to pick up new v1.1 directory/doc set) [online tool](https://www.xml-sitemaps.com/index.php?errmsg=wr&initurl=%2F). #17126
*Requirement*: All docs must be in the gh-pages branch (if something has not be uploaded yet, it wont get listed in the siteindex).  
*Instructions*: Run the [online tool](https://www.xml-sitemaps.com/index.php?errmsg=wr&initurl=%2F) by specifying the [http://kubernetes.io/](http://kubernetes.io/) URL. Then put the new sitemap.xml file in the gh-pages branch by replacing this existing [sitemap.xml](https://github.com/kubernetes/kubernetes/blob/gh-pages/sitemap.xml) file.
     - [x] Run "add sitemap" in [search console](https://www.google.com/webmasters/tools/sitemap-list?hl=en&siteUrl=http://kubernetes.io/#MAIN_TAB=0&CARD_TAB=-1). @brendandburns has access to this console.
 - [x] 3. (#16416) Expose v1.1 content for GA (will open a single PR to be merged for GA): 
     - [x] Expose drop-down version selector in the existing v1.0 doc set #16416
     - [x] Expose v1.0 banner (archive notice and redirect to v1.1) #16416
     - [x] Update links (so site defaults to v1.1): Update 404 page, update "Documentation" in main site horizontal nav bar, and all links in "http://kubernetes.io/gettingstarted/" #16416
     - [ ] ~~Create/add files for redirecting v1.0 urls to v1.1 (do this to maintain SEO/analytics and avoid negative impacts of changing all our URLs) [how to: .htaccess](http://httpd.apache.org/docs/2.0/howto/htaccess.html)~~  
This is not possible - [GitHub pages does not support .htaccess.html files](https://help.github.com/articles/redirects-on-github-pages/). The alternative they provide requires that each individual page include an automatic redirect which requires updates to the scripts to update the Jekyll "frontmatter" and would have the negative affect of disallowing anyone to view the v1.0 content (they would always get redirected). We will have to leave this as is for now so that users have the choice of viewing either v1.0 or v1.1.

**Comment by bgrant0607** [comment]
@RichieEscarez Is the drop-down version selector going to happen?

All the other changes appear to be fairly mechanical, and we could have PRs in place, ready to flip to v1.1.

**Comment by bgrant0607** [comment]
Most page titles are broken. They are specified like this:
```
title: "</strong>"
```
which either appears as `</strong>` in the browser tab title, or as the file name.

Only a handful are useful page descriptions:
```
grep -r "^title:" . | grep -v "</strong>"
./gs-localmachine.md:title: "Getting Started: Local-machine Solutions"
./gs-custom.md:title: "Getting Started: Custom Solutions"
./basicstutorials.md:title: "Quick Walkthrough: Kubernetes Basics"
./multi-tier.md:title: "Examples: Multi-tier Applications"
./api-ref.md:title: "Kubernetes API Reference"
./gs-turnkey.md:title: "Getting Started: Turn-key Solutions"
./setup-config.md:title: "Examples: Setup and Configuration"
./app-admin-detailed.md:title: "Application Administration: Detailed Walkthrough"
./deploy-clusters.md:title: "Examples: Deploying Clusters"
```

I assume some munger mis-munged these?

**Comment by bgrant0607** [comment]
Looks like #16170 added the bad titles. @krousey

**Comment by krousey** [comment]
This might be a side effect of populating those test pages from master instead of a release branch. I think the release mungers usually remove something before it gets to the gh-pages import tool.

**Comment by thockin** [comment]
Kris - are you on this?

**Comment by krousey** [comment]
@thockin Sorry I was just suggesting that that might be the case. If it is, then it's not a problem as we shouldn't never really release docs from the master branch. They should always come from a release branch, right?

**Comment by thockin** [comment]
If that is really the case, then I can buy it (though it would be nice to
a) prove it and b) be smarter :)

**Comment by krousey** [comment]
a) I'm looking into it now to be sure
b) i'll file an issue if part a goes as expected

**Comment by krousey** [comment]
Also worth noting that every file imported has this issue. The ones in @bgrant0607's list aren't imported.

**Comment by krousey** [comment]
a) confirmed. the `</strong>` comes from

```markdown
<strong>
The latest 1.0.x release of this document can be found
[here](http://releases.k8s.io/release-1.0/docs/api.md).

Documentation for other releases can be found at
[releases.k8s.io](http://releases.k8s.io).
</strong>
--
```

in file. The -- fools the importer into thing that the first header is `</strong>` and should be the title. The import.sh script in gh-pages strips this after importing them. They're already stripped in the release branch, so it shouldn't be an issue.

I guess the fix would be to somehow change import.sh to strip the release information before importing as well.

**Comment by bgrant0607** [comment]
Why can't we just import from the release branch, as intended?

**Comment by krousey** [comment]
Richie specifically requested a pull from master to test something. I guess
something the release branch doesn't have.

**Comment by RichieEscarez** [comment]
In chatting with caesarxuchao, we found that the alpha/beta extensions (new features for v1.1) are not in the release branch so we decided to import from master. Im not sure what the work around should be to pick up all the late incoming v1.1 docs (can those be pushed back into the v1.1 release?)?

The version selector button is in progress, will open the PR today along with a "flip the switch to v1.1" PR by EOD.

**Comment by RichieEscarez** [comment]
The following files are in gh-pages only (they have no use in master and act as landing pages that dynamically pull in and display the TOC from the nav menu) which is why the munger doesnt touch them:

```
./gs-localmachine.md:title: "Getting Started: Local-machine Solutions"
./gs-custom.md:title: "Getting Started: Custom Solutions"
./basicstutorials.md:title: "Quick Walkthrough: Kubernetes Basics"
./multi-tier.md:title: "Examples: Multi-tier Applications"
./api-ref.md:title: "Kubernetes API Reference"
./gs-turnkey.md:title: "Getting Started: Turn-key Solutions"
./setup-config.md:title: "Examples: Setup and Configuration"
./app-admin-detailed.md:title: "Application Administration: Detailed Walkthrough"
./deploy-clusters.md:title: "Examples: Deploying Clusters"
```

**Comment by bgrant0607** [comment]
cc @janetkuo

**Comment by bgrant0607** [comment]
We need to cherrypick whatever docs we need into 1.1.

**Comment by janetkuo** [comment]
Found and filed 2 gh-pages issues:
- gh-pages shows red rectangles in code #16391
- Download link doesn't display well in gh-pages #16393

**Comment by RichieEscarez** [comment]
FYI: ill work on #3 in this list later tonight.

**Comment by bgrant0607** [comment]
Thanks, @RichieEscarez

**Comment by bgrant0607** [comment]
We need to audit the docs that are in the 1.1 branch to ensure that all the docs we need have been cherrypicked.

**Comment by bgrant0607** [comment]
cc @caesarxuchao @nikhiljindal @thockin

**Comment by bgrant0607** [comment]
user-guide/kubectl-overview.md needs to be copied to release-1.1

**Comment by bgrant0607** [comment]
#15339 needs to be cherrypicked

**Comment by bgrant0607** [comment]
#15587 would be nice to have

**Comment by bgrant0607** [comment]
docs/user-guide/jobs.md is not up to date in release-1.1 (multiple PRs)

**Comment by bgrant0607** [comment]
It looks like we need to run hack/update-generated-docs.sh on the release branch

**Comment by bgrant0607** [comment]
#15453 needs to be cherrypicked

**Comment by bgrant0607** [comment]
And #14632

**Comment by bgrant0607** [comment]
And #15693

**Comment by bgrant0607** [comment]
#16043

**Comment by bgrant0607** [comment]
#15522

**Comment by bgrant0607** [comment]
#15790

**Comment by bgrant0607** [comment]
and docs/getting-started-guides/vagrant.md (multiple PRs)

**Comment by bgrant0607** [comment]
#13724

**Comment by bgrant0607** [comment]
The other differences are either insignificant, developer-oriented (so I'd expect them to read from github), misc getting-started guides, or changes that might not be in 1.1.

**Comment by bgrant0607** [comment]
There are some examples differences, including a few new examples, also, but none look critical.

**Comment by bgrant0607** [comment]
One more cherrypick: #16138

We need to:
1. Cherrypick/copy the PRs/docs mentioned above into release-1.1
2. Fix #16393
3. Run hack/update-generated-docs.sh in release-1-1
4. Do step (2) above (sitemap)
5. Run versionize docs again
6. Copy/munge release-1.1 docs to gh-pages

When we're ready to release 1.1:
Merge #16388 and #16416

**Comment by bgrant0607** [comment]
ETA - tomorrow (Thursday)

**Comment by bgrant0607** [comment]
PR #16461 includes all the cherrypicks except #15790, which had merge conflicts, #16138, which hasn't merged yet, and the files that are easier to copy than cherrypick: kubectl-overview, jobs.

**Comment by bgrant0607** [comment]
PR #16463 includes kubectl-overview, jobs.

**Comment by bgrant0607** [comment]
#16471 versionized docs.

**Comment by bgrant0607** [comment]
#16475 to fix #16393

**Comment by nikhiljindal** [comment]
Found one that you missed :)

Sent https://github.com/kubernetes/kubernetes/pull/16479 for #15832

**Comment by bgrant0607** [comment]
We should cherrypick #15981

**Comment by nikhiljindal** [comment]
Sent https://github.com/kubernetes/kubernetes/pull/16483 for #15981

**Comment by bgrant0607** [comment]
#16486 deleted the v1.1 docs from gh-pages and re-copied and munged them. That fixed the titles.

2 additional problems to fix here:
https://github.com/kubernetes/kubernetes/pull/16486#issuecomment-152082539

And, the download link doesn't work for files ending in `.in`.

**Comment by bgrant0607** [comment]
One (hopefully last) cherrypick: #16518

**Comment by mikedanese** [comment]
Also #16530

**Comment by bgrant0607** [comment]
And, now none of the navigation sidebar submenus expand, in either 1.1 or 1.0. :-(

**Comment by nikhiljindal** [comment]
https://github.com/kubernetes/kubernetes/pull/16540 to fix the navigation sidebar issue

**Comment by bgrant0607** [comment]
All of the issues mentioned above have been fixed and merged.

#16542 to change links to point to release branch instead of patch version tag.

**Comment by bgrant0607** [comment]
After we fix the links, we need to delete and re-copy the docs from release-1.1 to gh-pages again.

**Comment by bgrant0607** [comment]
Note that we found that the gh-pages munger (_tools/release_docs/main.go) is not idempotent. It expects the docs to be deleted from gh-pages and re-copied from the release branch.

**Comment by krousey** [comment]
@bgrant0607 _tools/import.sh should be doing this.

**Comment by bgrant0607** [comment]
_tools/import.sh doesn't work for me.
a) It doesn't actually work with /bin/sh. It needs bash.
b) I can't build the go tool due to lack of godeps

**Comment by bgrant0607** [comment]
#16567 to re-import

**Comment by bgrant0607** [comment]
Someone needs to regenerate the sitemap (step 2 in the PR description) after #16567 merges

**Comment by bgrant0607** [comment]
#16567 is merged

**Comment by bgrant0607** [comment]
Confirmed that red rectangles are gone

**Comment by bgrant0607** [comment]
The sitemap generator didn't pick up the v1.1 docs. It crawls links, and there aren't any yet.

Also, with just the 1.0 docs, it crawled 256 docs and has a max of 500, so we may not be able to crawl all the 1.1 docs, unless we omit the 1.0 docs, which apparently is what would happen once we merge the robots.txt change, since legacy wasn't crawled.

**Comment by bgrant0607** [comment]
Or it could just be that there are no links to legacy, I guess.

**Comment by janetkuo** [comment]
We should add `SITEMAP: http://kubernetes.io/sitemap.xml` to robot.txt (see http://www.advancedhtml.co.uk/robots-sitemaps.htm)

Tried some experiments on sitemap generator, here are the results:
- kubernetes.io returns 256 links
- kubernetes.io/v1.1 returns 286 links
- kubernetes.io/legacy returns 0 links

Conclusion: What's disallowed in robot.txt wouldn't be crawled by sitemap generator. We can change robot.txt and then re-generate the sitemap.

**Comment by janetkuo** [comment]
Tried adding a link to v1.1 docs in my http://janetkuo.github.io/kubernetes/, and sitemap generator can find those links now. It didn't pick up v1.1 docs in http://kubernetes.io because it's not referenced anywhere.

**Comment by nikhiljindal** [comment]
Next steps:

* @caesarxuchao is adding version selector from Richie's PR (the default will still be 1.0). With that we will start linking to v1.1 docs from kubernetes.io and hence 1.1 docs will start being indexed by crawlers.
* To launch 1.1 (whenever we are ready to do that):
   * Submit rest of Richie's PR to flip all links to point to 1.1 docs instead of 1.0 (@caesarxuchao )
   * Update robots.txt to disallow 1.0 (@janetkuo has a PR already)
   * Regenerate sitemaps.xml. It should now contain 1.1 links and should not contain 1.0 links. Update robots.txt to include this sitemaps.xml (@janetkuo)

**Comment by janetkuo** [comment]
Just in case sitemap generator failed to pick up the v1.1 link, create a hidden link to enable it. #16606

**Comment by janetkuo** [comment]
Sitemap generator now finds v1.1 links after #16606 is merged.

**Comment by janetkuo** [comment]
I merged the PR that updates robots.txt, regenerated sitemap.xml, and ran "resubmit sitemap" in [search console](https://www.google.com/webmasters/tools/sitemap-list?hl=en&siteUrl=http://kubernetes.io/#CARD_TAB=-1&MAIN_TAB=0).

**Comment by janetkuo** [comment]
Closing this since all todos are done.
